src/main/java/org/example/LessonDao.java

```diff
@@ -0,0 +1,110 @@
+package org.example;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LessonDao {
+    public void addHomework(Homework homework) throws SQLException {
```

Comment:
We suppose, that lesson dao is responsible only for managing the lessons. If you need to perform any operations on homework, it should be done in a separate class, like HomeworkDao, for instance. Having this in mind, do some refactoring to achieve that, please.